Bundle/PageBundle/Helper/PageHelper.php

```diff
@@ -175,6 +179,39 @@ public function renderPageByUrl($url, $locale, $layout = null)
 
             return $this->renderPage($page, $layout);
         } else {
```

Comment:
that is complicated 😰

Please refactor this part. An easy way to improve fast this part is to simply move it in a class named for example `ErrorHandler` which will handle the url, find or create the error, increaseCounter if needed and throw a NotFoundHttpException or return a redirection if it does exist.